lib/compiler/test/slotted-api.test.mjs

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-console */
+import { transform } from '@astrojs/compiler';
+
+const contents = `
+{Astro.slots.a && <div id="a">
```

Comment:
Could this be a Go test? Or is this better was a WASM test?

Reply:
Our go tests don't do a great job of testing `as: 'fragment'` behavior right now because they're handled in the WASM entrypoint.

I'd like to fix that but it's a follow-up IMO.